common/ontology.py

```diff
@@ -0,0 +1,56 @@
+import logging
+import random
+import time
+
+from ontoma.interface import OnToma
+from pandarallel import pandarallel
+
+ONTOMA_MAX_ATTEMPTS = 5
+pandarallel.initialize()
+
+
+def _ontoma_udf(row, ontoma_instance):
+    disease_name, disease_id = row['diseaseFromSource'], row['diseaseFromSourceId']
+    for attempt in range(1, ONTOMA_MAX_ATTEMPTS + 1):
+        # Try to map first by disease name (because that branch of OnToma is more stable), then by disease ID.
+        try:
+            mappings = ontoma_instance.find_term(query=disease_name, code=False)
+            if not mappings:
+                mappings = ontoma_instance.find_term(query=disease_id, code=True)
+            return [m.id_ot_schema for m in mappings]
+        except:
+            # If this is not the last attempt, wait until the next one
+            if attempt != ONTOMA_MAX_ATTEMPTS:
```

Comment:
You're right, it would cause an unnecessary delay.

I decided to restructure it in such a way that an exception is not caused at all, by adding an explicit additional check for whether the field is populated: https://github.com/opentargets/evidence_datasource_parsers/pull/94/commits/101c532e48752fbae904b34d3ecfe4797ee3b5d2